src/SFA.DAS.ApprenticeCommitments/Application/Commands/CreateApprenticeship/UpdateApprenticeshipCommand.cs

```diff
@@ -0,0 +1,83 @@
+    }
+
+    public class UpdateApprenticeshipCommandHandler : IRequestHandler<UpdateApprenticeshipCommand>
+    {
+        private readonly ApprenticeCommitmentsService _apprenticeCommitmentsService;
+        private readonly CommitmentsV2Service _commitmentsService;
+        private readonly TrainingProviderService _trainingProviderService;
+        private readonly ApprenticeLoginService _apprenticeLoginService;
+        private readonly CoursesService _coursesService;
+
+        public UpdateApprenticeshipCommandHandler(
+            ApprenticeCommitmentsService apprenticeCommitmentsService,
+            ApprenticeLoginService apprenticeLoginService,
+            CommitmentsV2Service commitmentsV2Service,
+            TrainingProviderService trainingProviderService,
+            CoursesService coursesService)
+        {
+            _apprenticeCommitmentsService = apprenticeCommitmentsService;
+            _apprenticeLoginService = apprenticeLoginService;
+            _commitmentsService = commitmentsV2Service;
+            _trainingProviderService = trainingProviderService;
+            _coursesService = coursesService;
+        }
+
+        public async Task<Unit> Handle(
+            UpdateApprenticeshipCommand command,
+            CancellationToken cancellationToken)
+        {
+            var (provider, apprenticeship, course) = await GetExternalData(command);
+            var id = Guid.NewGuid();
+
+            await _apprenticeCommitmentsService.ChangeApprenticeship(new ChangeApprenticeshipRequestData
+            {
+                ApprenticeshipId = command.ApprenticeshipId,
+                Email = command.Email,
+                EmployerName = apprenticeship.EmployerName,
+                EmployerAccountLegalEntityId = apprenticeship.AccountLegalEntityId,
+                TrainingProviderId = apprenticeship.ProviderId,
+                TrainingProviderName = string.IsNullOrWhiteSpace(provider.TradingName) ? provider.LegalName : provider.TradingName,
+                CourseName = course.Title,
+                CourseLevel = course.Level,
+                PlannedStartDate = apprenticeship.StartDate,
+                PlannedEndDate = apprenticeship.EndDate,
+                ApprovedOn = command.ApprovedOn,
+            });
+
+            return Unit.Value;
+        }
+
+        private async Task<(TrainingProviderResponse, CommitmentsApprenticeshipResponse apprenticeship, StandardApiResponse)>
+            GetExternalData(UpdateApprenticeshipCommand command)
+        {
+            var apprenticeship = await _commitmentsService.GetApprenticeshipDetails(
+                command.ApprenticeshipId);
+
+            var course = _coursesService.GetCourse(apprenticeship.CourseCode);
+            var provider = _trainingProviderService.GetTrainingProviderDetails(apprenticeship.ProviderId);
+
+            await Task.WhenAll(course, provider);
+
+            return (provider.Result, apprenticeship, course.Result);
+        }
+    }
+}
```

Comment:
I still believe it's worthwhile putting these particular classes into individual files within the usual command structure. What happened here is, is we ended up putting this code in a file in the wrong folder and none of us spotted it first code review. 

Consider what happens when CI take over the project, they will quickly scan down the Command sections and make an overall assumption, that there is no UpdateApprenticeshipCommand. 

I'm not a advocate of "all classes must be in separate files", but where there are common patterns across teams, we should follow them. If only for consideration of future devs

Reply:
I think we've had this discussion before :)  I like being able to F12 to the command handler directly from the controller.  I almost never want to know what the properties of the DTO are, I want to know what effect using the endpoint has on the system.

Yes, this update file was in the wrong folder.  I'm afraid I don't for one second buy the idea that if the classes were in two separate files in the wrong folder that would have been picked up any more that the single file was.

Reply:
Well think this is a discussion we need at a higher level. Dan Beavan might the best person to say what patterns we should and shouldn't use. I've corrected it in my branch and moved the files to the correct folder and removed email